File: 1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        set<int> st(arr.begin(), arr.end());
        unordered_map<int, int> mp;
        int rank = 1;
        for (int x : st) {
            mp[x] = rank;
            rank++;
        }
        for (int &x : arr) {
            x = mp[x];
        }
        return arr;
    }
};

File: LeetCode/1331-rank-transform-of-an-array/1331-rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        Set<Integer> st = new TreeSet<>();
        for (int x : arr) st.add(x);
        int rank = 1;
        Map<Integer, Integer> mp = new HashMap<>();
        for (int x : st) {
            mp.put(x, rank);
            rank++;
        }
        for (int i = 0; i < arr.length; i++) {
            arr[i] = mp.get(arr[i]);
        }
        return arr;
    }
}